### PR TITLE
Modernize Icon Playbook Kit

### PR DESCRIPTION
**Pull request by kskilt**
Modernize Icon kit to match new formatting

Additional Information:
Commit 5d69785 adjusts the Enum specs to coerce array into strings for comparison. This allows the default to be nil.

### CODE REVIEW

File: app/pb_kits/playbook/pb_icon/_icon.html.erb [hunk 1]
```diff
@@ -1,7 +1,4 @@
-<% if !object.icon_class.nil? %>
-  <%= content_tag(:i, nil,
-      id: object.id,
-      data: object.data,
-      class: object.classname(object.kit_class),
-      aria: object.aria) %>
```
Comment (garettarrowood) [inline app/pb_kits/playbook/pb_icon/_icon.html.erb]: We want to keep `aria` in there. It's a base prop.

Comment (kskilt) [reply app/pb_kits/playbook/pb_icon/_icon.html.erb]: Added back in

File: spec/support/playbook/rspec.rb [hunk 2]
```diff
@@ -1,5 +1,7 @@
 # frozen_string_literal: true
 
+require "pry"
```
Comment (garettarrowood) [inline spec/support/playbook/rspec.rb]: You will have to remove this too. It will likely fail your CI (or I hope it would).

Comment (kskilt) [reply spec/support/playbook/rspec.rb]: It did fail, and I thought we ran `git co` on this file. Good catch!

**Comment by kskilt** [comment]
> LGTM. Did you smoke test this?

Yup, threw it up on localhost:8080 and compared to prod. Everything looked the same

**Pull request merged**